### ISSUE TITLE
Add missing compatibility

### ISSUE THREAD

**Issue by CobaltWolf** [CobaltWolf/Bluedog-Design-Bureau#101]
None of the new parts have compatibility made yet (save for RP configs), so things like Tweakscale still need to be done. This includes many parts from the last release.

**Comment by CobaltWolf** [comment]
@jsolson @VenomousRequiem we should try and take care of this tonight if we can. I'll look into the MOS parts I suppose. There was a lot of stuff in the last two releases that didn't have stuff like Tweakscale configs, so we'll have to comb through a bit.

**Comment by jsolson** [comment]
I could probably write an MM config that will find all the bluedog* parts that do not have ModuleTweakscale or whatever and change the titles to make identifying them easier. Remind me on the weekend if I don't do it in the next few days.

**Comment by CobaltWolf** [comment]
https://www.facebook.com/nixcraft/posts/1353058694707505

**Comment by jsolson** [comment]
There's over 70 parts scattered around missing a tweakscale config. This is going to be a job and a half.

**Comment by CobaltWolf** [comment]
@jsolson do you have a print out or something to make it easier for someone to take care of that?

**Comment by jsolson** [comment]
I uploaded an MM patch in the tweakscale folder to identify parts missing tweakscale. You need to rename it from txt to cfg. Changes the title and moves the parts to the Control category.

**Comment by Cowboy-WY** [comment]
Not sure it is easy enough to explain, but if you can explain to me what needs to be added and where, I could put some time toward this.  I've not used tweakscale.

**Comment by jsolson** [comment]
Get it. It's a handy tool I often forget I have.

Our configs are not the best examples for tweakscale since you can do more with it than we're doing. To get you started look at how the tanks are done. You add a MODULE[TweakScale], the type = BluedogStack line give it support for our off sizes (defined in tweakscale_Scales.cfg), and the defaultScale = 0.9375 would be set to the native size of the part.

Some parts have a TWEAKSCALEBEHAVIOR added. Engines need special handling because you're not just making it bigger, tweakscale needs to adjust the thrust and things.

Science should also have a behavior. Look at Coatle's probes plus configs for examples, I did those. I think all the science behavior does is make it more expensive, rather than cheaper when you make the part smaller. There may be other behaviors we should be using. I'm no tweakscale expert.

Some parts are not "stack" parts that fit cleanly into 1.24, 2.5 etc sizes. Solar panels, pretty much anything radial. Those might be type = free, or type = free_square (I think). Look around for some examples. Tweakscales own files for the squad parts are probably great examples.

**Comment by Cowboy-WY** [comment]
Thanks,  I'll see what I can figure out and run a few changes by you before I start updating them in bulk.

**Comment by CobaltWolf** [comment]
I believe solar panels use free_square.

@Cowboy-WY I saw the changes you uploaded. Typically crewed parts don't receive Tweakscale, as they don't work right with it. I'm not sure how game breaking it is, I just know it's a no-no.

Fairing bases also cannot be made tweakscale compatible.

**Comment by Cowboy-WY** [comment]
I wondered about the crewed parts, but saw the Hermes pod had it configured.  Want me to remove all crewed part tweakscale configs?

**Comment by Cowboy-WY** [comment]
Tweakscale should be complete with #129 except for Saturn.  I opened a new issue for that to re-visit before release.

I did remove tweakscale from all crewed parts.

**Comment by jsolson** [comment]
I'm not sure it's true anymore that crewed parts don't work. They seem to have explicit support for it now.

Fairing bases don't work though - I confirmed that myself when I was messing with these last. PFairings have built in scaling support though.

**Comment by Cowboy-WY** [comment]
Let me know if you want the crewed parts re-added, would only take a few minutes after #129 is merged.

**Comment by jsolson** [comment]
Let's wait and see what CW has to say about crewed parts. The Tweakscale config for Squad seems to be using stack_square for crewed parts.

I assume your editing through the website's editor? When your doing a bunch of edits in one folder, try editing the files on your local disk, the use the upload files button for that folder in your branch to upload them all at once. Git will figure out that your changing things, not adding new files. You'll only have one commit that way. Using one of the windows clients is the best way but that's a little more advanced.

When I'm using a branch I generally delete it after it's been merged to the main branch, because I haven't figured out how to keep branches synched. You branch will fall behind the main branch as other peoples commits get added.

**Comment by Cowboy-WY** [comment]
I think using stack_square would require a new type to get the custom sizes.

I have been using the website editor.

Thanks a bunch for your advice.  I'm a complete noob with this github stuff.  :)

I'm learning a little each time and I just figured out how to update my branch from this master last night, but it makes everything seem messy.  I do intend on deleting the branches each time from now on.  I will also try to figure out the offline editing a bit over the next few things I do.

**Comment by CobaltWolf** [comment]
Are we good on this? Do any of our other compatibility things need to be updated?

**Comment by jsolson** [comment]
There's AR, CTT, and RT. I think they have maintainers.

CryoEngines. I've got that.

Hullcam. Is that mod still alive?

RealChute. Needs updating.

And TweakScale which appears to be under control. Though it's an ongoing
issue at the rate you model parts.

**Comment by CobaltWolf** [comment]
RT is/was updated by @Cowboy-WY. 
AR and CTT were by @rasta013 , I haven't seen him in a couple weeks.
Hullcam is in flux, don't worry about it.
Realchute needs a small update.
